GomokuServer/src/GomokuServer.Application/Games/Commands/Abstract/Undo.cs

```diff
@@ -0,0 +1,34 @@
+﻿
+namespace GomokuServer.Application.Games.Commands.Abstract;
+
+public abstract record UndoCommand : ICommand<UndoResponse>
+{
+	[Required]
+	public required string GameId { get; init; }
+
+	[Required]
+	public required string PlayerId { get; init; }
+}
+
+public abstract class UndoCommandHandler<TRequest>(IGamesRepository _gamesRepository)
+	: ICommandHandler<TRequest, UndoResponse>
+		where TRequest : UndoCommand
+{
+	public async Task<Result<UndoResponse>> Handle(TRequest request, CancellationToken cancellationToken)
+	{
+		var getGameResult = await _gamesRepository.GetAsync(Guid.Parse(request.GameId));
+
+		return getGameResult.Bind(game =>
+		{
+			var undoResult = game.Undo(request.PlayerId);
+
+			return undoResult.IsValid
+				? Result.Success(new UndoResponse()
+				{
+					RemovedTile = TileDto.FromDomainModel(undoResult.RemovedTile),
+					PreviouslyPlacedTile = TileDto.FromDomainModel(undoResult.PreviouslyPlacedTile)
+				})
+				: Result.Invalid(new ValidationError(undoResult.ErrorDetails));
+		});
+	}
```

Comment:
For commands (and queries) we place both (query/command and handler) in one file for better debugging, because you can't put breakpoint on mediator Send method and press F11 if you want to debug some handler logic. Mediator will resolve handler in it's internals and you will step into mediator's code. So, to avoid this problem and have better debugging experience, we can click on query/command in IDE, go to file where handler is located as well and put breakpoint there. So, maybe sounds like weird reason to put something in one file, but mediator is not trivial in a way, how it resolves handler for specific request.

Reply:
But do we need to put breakpoint on mediator, if you can just wrap it with logger in development mode?

Reply:
I often use breakpoints if I need to debug something, I am not that advanced to find all problems with logger 😆
But anyway, even without discussion about debugging practice, because of how Mediator library works, this relation between query/command and handler is transparent. Explicitly with new operator we create only command/query, but mediator does it work to find correct handler (using reflection by reading handler generic type). So, taking this into account, I decided, that convention command/query with handler in one file is very good. You can clearly understand, that this things are related. 

You raise good questions, I am thinking for a long time to write some README for server project with naming convention description, libraries usage and architecture pattern description. 

I created issue - https://github.com/ligomoku/gomoku/issues/238
Please add things, what else you think maybe be important in this issue

Reply:
Was just wondering about the logging setup in the mediator. Since we're using console.debug on the frontend side, maybe it makes sense to do similar logging in the mediator too? Would give us better Sentry logs. What if we wrapped the mediator with logging in general? What do you think?

Reply:
For conventions let's please use wiki -> https://github.com/ligomoku/gomoku/wiki